tests/backends/test_cursors.py

```diff
@@ -0,0 +1,55 @@
+import unittest
+
+from django.db import connection
+from django.test import TestCase
+
+
+class LoggingCursorMixin:
+    def __init__(self, bucket, *args, **kwargs):
+        super().__init__(bucket, *args, **kwargs)
+        self.bucket = bucket
+
+    def execute(self, sql, *args, **kwargs):
+        self.bucket.append(sql)
+        super().execute(sql, *args, **kwargs)
+
+
+@unittest.skipUnless(connection.vendor == 'mysql', 'MySQL specific test.')
+class MySQLCursorOptionsTestCase(TestCase):
+    try:
+        from MySQLdb.cursors import Cursor
+
+        class MySQLLoggingCursor(LoggingCursorMixin, Cursor):
+            pass
+    except ImportError:
+        pass
+
+
+@unittest.skipUnless(connection.vendor == 'oracle', 'Oracle specific test.')
+class OracleCursorOptionsTestCase(TestCase):
+    try:
+        import cx_Oracle as Database
+
+        class OracleLoggingCursor(LoggingCursorMixin, Database):
```

Comment:
If `cx_oracle` is installed, there's an error:
```
 File "/home/tim/code/django/tests/backends/test_cursors.py", line 33, in OracleCursorOptionsTestCase
    class OracleLoggingCursor(LoggingCursorMixin, Database):
TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases
```
Also, is this file doing anything useful? I don't see any test methods.